core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -1068,8 +1059,7 @@ class DynamicListenerConfig(server: KafkaBroker) extends BrokerReconfigurable wi
         listenersToMap(newConfig.effectiveAdvertisedBrokerListeners))) {
       verifyListenerRegistrationAlterationSupported()
       server match {
-        case kafkaServer: KafkaServer => kafkaServer.kafkaController.updateBrokerInfo(kafkaServer.createBrokerInfo)
-        case _ => throw new RuntimeException("Unable to handle non-kafkaServer")
+        case _ => throw new RuntimeException("Unable to handle reconfigure")
```

Comment:
If we always throw this exception, we probably don't need to do `verifyListenerRegistrationAlterationSupported` before that and we may be able to delete that method if it's not used anywhere else.

Reply:
Also, we probably should include more context: it looks like we don't allow dynamic reconfiguration of listener registrations.

@cmccabe @jsancio Is this documented as part of the zk to kraft migration?

Reply:
We should remove `SocketServerConfigs.ADVERTISED_LISTENERS_CONFIG` from `ReconfigurableConfigs` as well.

https://github.com/apache/kafka/blob/a628d9bc4d82ddb63ca00f7bcd88a6a6788eae7f/core/src/main/scala/kafka/server/DynamicBrokerConfig.scala#L912

That can disallow users to configure advertised listeners dynamically.

Reply:
I open a Jira to test all dynamic config which can't dynamic configure in Kraft, and only find `ADVERTISED_LISTENERS_CONFIG` can't, If we addressed it on this PR, I will close [Jira](https://issues.apache.org/jira/browse/KAFKA-18403) and https://github.com/apache/kafka/pull/18390

Reply:
> There's a bunch of other ZooKeeper related logic in this class so I was planning to address them together in a follow up issue. Deleting KafkaServer enables us to start removing other classes that depend on this one.

I'm ok to merge this and then address comments in the follow-up, because there are many cleanup blocked by KafkaServer

Reply:
I'm ok with that, but let's file a JIRA so we don't lose track.

Reply:
I care particularly about the error messages and such things since they are not easy to find afterwards.

Reply:
I opened https://issues.apache.org/jira/browse/KAFKA-18405 with details about this issue.